### PR TITLE
[fix]:[CDS-58444]: Revert EKS v2 API changes for fetching cluster details

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
We are reverting the changes currently made for EKS V2 API. It will be added for all the k8s infra together as part of another epic. https://harness.atlassian.net/browse/CDS-58901

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46240)

### CODE REVIEW

**Comment by lovish1999** [comment]
trigger messagecheck

**Comment by lovish1999** [comment]
trigger ut3

**Comment by lovish1999** [comment]
trigger ut2

**Comment by lovish1999** [comment]
trigger ut0

**Comment by lovish1999** [comment]
trigger ut1

**Review by abhinavsingh1196 — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46240)* status: 3 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): k8s-team.

**Comment by markoharness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/aws/resources/AwsHelperResource.java` line 410 at r1](https://reviewable.io/reviews/harness/harness-core/46240#-NSRwLQ_3NXHjpkcliPf:-NSRwLQ_3NXHjpkcliPg:bfe9e7o) ([raw file](https://github.com/harness/harness-core/blob/308dc645170dad408863379bc67185d0c8afec9a/120-ng-manager/src/main/java/io/harness/ng/core/aws/resources/AwsHelperResource.java#L410)):*
> ```Java
>           NGCommonEntityConstants.ENVIRONMENT_KEY) String envId,
>       @Parameter(description = NGCommonEntityConstants.INFRADEF_PARAM_MESSAGE) @QueryParam(
>           NGCommonEntityConstants.INFRA_DEFINITION_KEY) String infraDefinitionId) {
> ```

Should envId and infraDefinitionId params be mandatory?



_Code quote:_
```Java
      @Parameter(description = NGCommonEntityConstants.ENV_PARAM_MESSAGE) @QueryParam(
          NGCommonEntityConstants.ENVIRONMENT_KEY) String envId,
      @Parameter(description = NGCommonEntityConstants.INFRADEF_PARAM_MESSAGE) @QueryParam(
          NGCommonEntityConstants.INFRA_DEFINITION_KEY) String infraDefinitionId) {
```

**Review by markoharness — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46240)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46240)* status: 3 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/aws/resources/AwsHelperResource.java` line 410 at r1](https://reviewable.io/reviews/harness/harness-core/46240#-NSRwLQ_3NXHjpkcliPf:-NSTWbe85opv9VLzI444:b-73oc2m) ([raw file](https://github.com/harness/harness-core/blob/308dc645170dad408863379bc67185d0c8afec9a/120-ng-manager/src/main/java/io/harness/ng/core/aws/resources/AwsHelperResource.java#L410)):*
<details><summary><i>Previously, markoharness (Marko Lukic) wrote…</i></summary><blockquote>

Should envId and infraDefinitionId params be mandatory?
</blockquote></details>

No because we are only using them for cases when we do not have connector value with us. There are cases when UI may not be able to pass env ID, but still the code with work. Later on in the getResolvedFieldValue method, we are finding envID based on fqn path & runtime yaml, so even if its null and required later on, the code can do it. 
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/aws/resources/AwsHelperResource.java` line 440 at r1](https://reviewable.io/reviews/harness/harness-core/46240#-NSRwZL45UsQRYnzhTcw:-NSTXCyw2x93Iva7GL2d:b-8nwhi) ([raw file](https://github.com/harness/harness-core/blob/308dc645170dad408863379bc67185d0c8afec9a/120-ng-manager/src/main/java/io/harness/ng/core/aws/resources/AwsHelperResource.java#L440)):*
> No because we are only using them for cases when we do not have connector value with us. There are cases when UI may not be able to pass env ID, but still the code with work. Later on in the getResolvedFieldValue method, we are finding envID based on fqn path & runtime yaml, so even if its null and required later on, the code can do it.

**Review by markoharness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46240)* status: 3 file(s) to review.

**Comment by lovish1999** [comment]
trigger review

File: 120-ng-manager/src/main/java/io/harness/ng/core/aws/resources/AwsHelperResource.java [hunk 1]
```diff
@@ -451,9 +451,10 @@ public ResponseDTO<List<String>> getEKSClusterNames(@QueryParam("awsConnectorRef
     if (isEmpty(awsConnectorRef) && spec != null) {
       awsConnectorRef = spec.getConnectorReference().getValue();
     }
+    String serviceId = null;
```
Comment (abosii-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/aws/resources/AwsHelperResource.java:454]: We can just pass null, don't need this temp var

**Review by abosii-harness — APPROVED** [review]

**Review by markoharness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46240)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by namantalaycha10 — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46240)* status: :shipit: complete! ✓

**Comment by lovish1999** [comment]
trigger review

**Pull request merged**